modules/_labs/pagination/react/README.md

```diff
@@ -14,82 +74,555 @@ yarn add @workday/canvas-kit-labs-react-pagination
+        <Pagination.StepToPreviousButton aria-label="Previous" />
+        <Pagination.PageList>
+          {({state}) =>
+            state.range.map(pageNumber => (
+              <Pagination.PageListItem key={pageNumber}>
+                <Pagination.PageButton aria-label={`Page ${pageNumber}`} pageNumber={pageNumber} />
+              </Pagination.PageListItem>
+            ))
+          }
+        </Pagination.PageList>
+        <Pagination.StepToNextButton aria-label="Next" />
+        <Pagination.JumpToLastButton aria-label="Last" />
+        <Pagination.GoToForm>
+          <Pagination.GoToTextInput />
+          <Pagination.GoToLabel>{({state}) => `of ${state.lastPage} results`}</Pagination.GoToLabel>
+        </Pagination.GoToForm>
+      </Pagination.Controls>
+      <Pagination.AdditionalDetails shouldHideDetails>
+        {({state}) => `${state.rangeMin}-${state.rangeMax} of ${state.lastPage} results`}
+      </Pagination.AdditionalDetails>
+    </Pagination>
+  );
+};
+```
+
+### Advanced Usage
+
+Below are examples for more advanced / custom usage of these components. We expect most people won't
+need to read this section, but if you're needing to go beyond our basic examples or just curious,
+feel free to explore this section. While these examples are not exhaustive, they provide additional
+insight into what's possible.
+
+#### Controllable State Using Pagination Model
+
+If you want the `Pagination` component to handle its state and actions internally and hook into page
+change events, the basic usage examples above should be sufficient. However, if you need to control
+the state externally, or if you want more control over the model, you'll need to use a controllable
+state pattern by using the `Pagination` model. You can create the model with the
+`usePaginationModel` hook provided:
+
+```tsx
+const model = usePaginationModel({
+  lastPage={100}
+  initialCurrentPage={6}
+  rangeSize={3}
+  onPageChange={pageNumber => console.log(pageNumber)}
+});
+```
+
+Notice that the hook takes the same props as the `Pagination` component (with the exception of
+`aria-label` which isn't needed to build the model). This allows you to quickly convert your
+implementation to this pattern if needed without a major refactor. You can then supply this model to
+the `Pagination` component.
+
+```tsx
+const ControllablePagination = () => {
+  // create the model
+  const model = usePaginationModel({
+  lastPage={100}
+  initialCurrentPage={6}
+  rangeSize={3}
+  onPageChange={pageNumber => console.log(pageNumber)}
+});
+
+  return (
+    <Pagination aria-label="Pagination" model={model}>
+      {/* child components go here */}
+    </Pagination>
+  );
+}
+```
+
+Notice that when providing a model, you no longer need to provide the other props (other than
+`aria-label`). Let's look at a complete example where you might want to use this controllable
+pattern.
+
+First, we'll create a table that will fetch some mock data:
+
+```tsx
+```
```

Comment:
> This example intentionally left blank

Reply:
It's...uh... the essay portion of the exam. Once they've read the docs, they should be able to fill in their own example here, and I'll grade it. 😉